setup.py

```diff
@@ -1,186 +1,94 @@
 
-# try bootstrapping setuptools if it doesn't exist
-try:
-    import pkg_resources
+import versioneer
+
+###############################################################################
+# Key Values that Change Each Release
+###############################################################################
+SETUP_REQUIREMENTS = {'numpy': '1.11',  # released March 2016
+                      'scipy': '0.18',  # released July 2016
+                      }
+
+REQ_NOT_MET_MSG = """
+{0} is installed but older ({1}) than required ({2}). You must manually
+upgrade {0} before installing or install into a fresh virtualenv.
+"""
+for key in SETUP_REQUIREMENTS:
+    import importlib
+    from distutils.version import LooseVersion
+    req_ver = LooseVersion(SETUP_REQUIREMENTS[key])
     try:
-        pkg_resources.require("setuptools>=0.6c5")
-    except pkg_resources.VersionConflict:
-        from ez_setup import use_setuptools
-        use_setuptools(version="0.6c5")
-    from setuptools import setup, Command, find_packages
-    _have_setuptools = True
-except ImportError:
-    # no setuptools installed
-    from distutils.core import setup, Command
-    _have_setuptools = False
+        mod = importlib.import_module(key)
+        ver = LooseVersion(mod.__version__)
+        if ver < req_ver:
+            raise RuntimeError(REQ_NOT_MET_MSG.format(key, ver, req_ver))
+    except ImportError:
+        pass
+    except AttributeError:
+        raise RuntimeError(REQ_NOT_MET_MSG.format(key, ver, req_ver))
+
+INSTALL_REQUIREMENTS = SETUP_REQUIREMENTS.copy()
+INSTALL_REQUIREMENTS.update({'pandas': '0.19',  # released October 2016
+                             'patsy': '0.4.0',  # released July 2015
+                             })
+
+CYTHON_MIN_VER = '0.24'  # released Apr 2016
 
-if _have_setuptools:
-    setuptools_kwargs = {"zip_safe": False}
-else:
-    setuptools_kwargs = {}
-    if sys.version_info[0] >= 3:
-        sys.exit("Need setuptools to install statsmodels for Python 3.x")
+SETUP_REQUIRES = [k + '>=' + v for k, v in SETUP_REQUIREMENTS.items()]
+INSTALL_REQUIRES = [k + '>=' + v for k, v in INSTALL_REQUIREMENTS.items()]
 
+EXTRAS_REQUIRE = {'build': ['cython>=' + CYTHON_MIN_VER],
+                  'develop': ['cython>=' + CYTHON_MIN_VER],
+                  'docs': ['sphinx', 'nbconvert', 'jupyter_client',
+                           'ipykernel', 'matplotlib', 'nbformat', 'numpydoc',
+                           'pandas-datareader']}
 
-curdir = os.path.abspath(os.path.dirname(__file__))
-README = open(pjoin(curdir, "README.rst")).read()
-CYTHON_EXCLUSION_FILE = 'cythonize_exclusions.dat'
+###############################################################################
+# Values that rarely change
+###############################################################################
 
 DISTNAME = 'statsmodels'
 DESCRIPTION = 'Statistical computations and models for Python'
+with open(pjoin(os.getcwd(), 'README.rst')) as readme:
```

Comment:
Is the getcwd() robust to running from outside this directory?

Reply:
Probably not